🚩PR: Various QA fixes of stable

elsoazemelet opened the pull request:
Closes #567 and #566 and #568

The issues fixes and features implemented:

- [x] In virtual module mode the clear/discard/store functionalities should work as with normal modules. Discard resets the module configuration to the last stored state, store stores the current state, and clear resets to the default configuration. 
- [x] Virtual Modules can be added properly on its modal dialog, without the dialog closing when selecting module type from the dropdown list.
- [x] Virtual modules should have their proper default configuration
- [x] In normal module mode, the discard, store, and clear should feedback and work as intended. (check logger messages)
- [x] Disconnecting modules during running processes (e.g.: during page store, profile load, etc.) should not break editor.

elsoazemelet committed:
Fixed device lost rejection

elsoazemelet committed:
Fixed mouse handling for modal base, and fixed event panel naming

elsoazemelet committed:
Updated dependency for grid-protocol

narayb commented:
Loading a (possibly faulty) button preset on a virtual TEK2 produces error:
![kép](https://github.com/intechstudio/grid-editor/assets/74784772/2180a6bc-f6fc-4dcf-a307-05a8626da1c2)

SukuWc commented on src/renderer/runtime/engine.store.ts:
```diff
@@ -157,17 +159,20 @@ function createWriteBuffer() {
-          ) {
-            incoming.filter.class_parameters["LASTHEADER"] = id;
+          let id = undefined;
+          try {
```
Set id in .then section and  reject in .catch section. TryCatch block is not needed, just await sendDataTo.Grid(incoming.descr).then(e){...}.catch(e){reject(e)}

elsoazemelet replied:
Fixed. I reorganized this part of the code for the better understanding of the process, including that this try/catch mechanism was promisified. Please validate if it is the correct way! Thank you for the feedback!

SukuWc commented on src/renderer/serialport/serialport.js:
These changes are welcome!

elsoazemelet committed:
Fixed QA and CR issue of loading possilby faulty config and removed try catch

elsoazemelet commented:
> Loading a (possibly faulty) button preset on a virtual TEK2 produces error: ![kép](https://private-user-images.githubusercontent.com/74784772/299362909-2180a6bc-f6fc-4dcf-a307-05a8626da1c2.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDYxNzQ0NDksIm5iZiI6MTcwNjE3NDE0OSwicGF0aCI6Ii83NDc4NDc3Mi8yOTkzNjI5MDktMjE4MGE2YmMtZjZmYy00ZGNmLWEzMDctMDVhODYyNmRhMWMyLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAxMjUlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMTI1VDA5MTU0OVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPThiMzc2ZTBjNmQ1NmQ1NDFkYWM0Nzc5YjRhNjYwMTk3MGFhNzJiZDJiNWQzZmYzOGFiYTg5NDRkNjkxY2ZjNGUmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.SvK2pKP5gK0rp_Sg3pdik3w5bN9ytDPqWwjsY7yEB80)

@narayb the reason of the error is that this button preset has 4 event configurations on it. Possibly the reason is because it is actually an encoder preset? Not sure, but fixed so the unknown event is not transferred to grid. @SukuWc  is this valid? If I remember correctly, this feature was implemented before, but I accidentally cut it out during refactor?

narayb commented:
`Disconnecting modules during running processes (e.g.: during page store, profile load, etc.) should not break editor.`

The above works well, but also allows for the loading of half-profiles which result in broken functionality on the Grid side, without much indication of this happening on the Editor side.

Logger should show that an unexpected disconnect occurred during store and for a successful store, the user should redo the process.

This way we warn the user on the software side as well.

elsoazemelet committed:
Fixed QA issue of feedback

elsoazemelet commented:
> `Disconnecting modules during running processes (e.g.: during page store, profile load, etc.) should not break editor.`
> 
> The above works well, but also allows for the loading of half-profiles which result in broken functionality on the Grid side, without much indication of this happening on the Editor side.
> 
> Logger should show that an unexpected disconnect occurred during store and for a successful store, the user should redo the process.
> 
> This way we warn the user on the software side as well.

Fixed the feedback message. I do believe that there was already a logging of the error of store/discard/clear when disconnecting, but the now changed feedback message is more prominent.

narayb commented:
Feedback message tested and working!
